src/components/OHAEditor/index.js

```diff
@@ -276,7 +276,27 @@ export default ({
                 changeSingleSampleOHA(
                   setIn(singleSampleOHA, ["taskOutput", relativeIndex], output)
                 )
-                onChangeOHA(newOHA)
+
+                let sendOnChange = true
+                if (
+                  oha.taskOutput &&
+                  newOHA.taskOutput &&
+                  oha.taskOutput[singleSampleOHA.sampleIndex] !== null
+                ) {
+                  const o1 = oha["taskOutput"][singleSampleOHA.sampleIndex]
+                  const o2 = newOHA["taskOutput"][singleSampleOHA.sampleIndex]
+                  if (
+                    oha.interface.type === "image_segmentation" &&
+                    o1.keys.length === o2.keys.length
+                  ) {
+                    if (JSON.stringify(o1) === JSON.stringify(o2)) {
```

Comment:
use `isEqual` from the npm module `lodash`, comparing the stringification of objects is unreliable because the key order can change.